qiskit/compiler/transpiler.py

```diff
@@ -512,6 +512,8 @@ def _parse_initial_layout(initial_layout):
         return initial_layout
     if isinstance(initial_layout, dict):
         return Layout(initial_layout)
+    if isinstance(initial_layout, range):
+        return list(initial_layout)
     if all(phys is None or isinstance(phys, Qubit) for phys in initial_layout):
         return Layout.from_qubit_list(initial_layout)
```

Comment:
I think at this point we could just do `initial_layout = list(initial_layout)` to handle any iterable?  At worst it'll make a copy of an existing list, but the list isn't going to be large, so it's not a big deal.

Reply:
Sure thing! I've added the test and release note, and I agree that we can just use `initial_layout = list(initial_layout)`. I made that adjustment.